rio/srcrobot/subsystems/Shooter.py

```diff
@@ -38,21 +38,33 @@ def __init__(self):
         self.VelocityControl = VelocityVoltage(0)
         self.VoltageControl = VoltageOut(0)
         self.percentOutput = DutyCycleOut(0)
-        
+
+        self.currentShotVelocity = 0.0
+
+    def setShooterVelocity(self, velocity):
+        self.topShooter.set_control(self.VelocityControl.with_velocity(Conversions.MPSToRPS(velocity,  0.101 * math.pi)))
+        self.bottomShooter.set_control(self.VelocityControl.with_velocity(Conversions.MPSToRPS(velocity,  0.101 * math.pi)))
+
+        self.currentShotVelocity = Conversions.MPSToRPS(velocity,  0.101 * math.pi)
+
+    def neutralizeShooter(self):
+        self.topShooter.set_control(self.VoltageControl.with_output(0))
+        self.bottomShooter.set_control(self.VoltageControl.with_output(0))
+
+    def idle(self):
+        return self.run(self.setShooterVelocity(Constants.ShooterConstants.kAmpShootSpeed))    
+    
     def shoot(self):
-        # return InstantCommand(lambda: self.bottomShooter.set_control(self.VelocityControl.with_velocity(Conversions.MPSToRPS(Constants.ShooterConstants.kPodiumShootSpeed,  0.101 * math.pi)))).alongWith(
-        #     InstantCommand(lambda:self.topShooter.set_control(self.VelocityControl.with_velocity(Conversions.MPSToRPS(Constants.ShooterConstants.kPodiumShootSpeed,  0.101 * math.pi))))
-        # )
-        return InstantCommand(lambda: self.bottomShooter.set_control(self.VelocityControl.with_velocity(Conversions.MPSToRPS(Constants.ShooterConstants.kPodiumShootSpeed,  0.101 * math.pi)))).alongWith(
-            InstantCommand(lambda:self.topShooter.set_control(self.VelocityControl.with_velocity(Conversions.MPSToRPS(Constants.ShooterConstants.kPodiumShootSpeed,  0.101 * math.pi))))
-        )
+        return self.run(lambda: self.setShooterVelocity(Constants.ShooterConstants.kPodiumShootSpeed))
 
     def shootReverse(self):
-        return InstantCommand(lambda: self.bottomShooter.set_control(self.VelocityControl.with_velocity(Conversions.MPSToRPS(-Constants.ShooterConstants.kPodiumShootSpeed,  0.101 * math.pi)))).alongWith(
-            InstantCommand(lambda:self.topShooter.set_control(self.VelocityControl.with_velocity(Conversions.MPSToRPS(-Constants.ShooterConstants.kPodiumShootSpeed,  0.101 * math.pi))))
-        )
+        return self.run(lambda: self.setShooterVelocity(-Constants.ShooterConstants.kPodiumShootSpeed))
+    
+    def isShooterReady(self):
+        topShooterReady = abs(abs(self.topShooter.get_velocity().as_supplier()) - self.currentShotVelocity) < 0.5
```

Comment:
I will admit that I am still learning this new API too, but I think this is not quite the right way to get the current velocity. The as_supplier() call returns a lambda and not the value. The lambda refreshes (non-blocking) and returns the value. I think the abs function will choke on a lambda. Also, this points to an interesting option. You should be able to call get_velocity().as_supplier() just once in initialization and then repeatedly use the lambda here.